### DISCUSSION TITLE
Partial rule evaluation / subcondition

### DISCUSSION THREAD

**Discussion by marek-kuticka** [hyperjumptech/grule-rule-engine#376] [Q&A]
Good evening.

if I have several rules including the same "subrule / subcondition" .. is there a way to remember, that the certain subcondition was already evaluated for a specific fact?

`rule 1:
when
 Person.Status == "single" && Person.Salary > 20000`

and then

`rule2:
when
Person.Status == "single" && Person.assets < 30000`

I'd like the evaluation of first subcondition to be remembered and re-used when evaluating the same fact using same subcondition.

is this possible in current GRULE engine? if not, could you imagine having this implemented?

many thanks

Marek

**Comment by newm4n** [comment]
As you can read here, https://github.com/hyperjumptech/grule-rule-engine/blob/master/docs/en/RETE_en.md#what-is-inside-grules-working-memory

The expression `erson.Status == "single"` will be remembered and its evaluation result.
